pkg/virtual/framework/wrappers/rbac/merging.go

```diff
@@ -0,0 +1,469 @@
+		item, err := lister.GetWithContext(ctx, name)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				errorHolder = err
+			}
+		} else {
+			return item, nil
+		}
+	}
+	return nil, errorHolder
+}
+
+func MergedClusterRoleInformer(informers []rbacinformers.ClusterRoleInformer) rbacinformers.ClusterRoleInformer {
+	return &mergedClusterRoleInformer{
+		informers: informers,
+	}
+}
+
+var _ rbacinformers.ClusterRoleInformer = (*mergedClusterRoleInformer)(nil)
+var _ rbaclisters.ClusterRoleLister = (*mergedClusterRoleLister)(nil)
+
+type mergedClusterRoleInformer struct {
+	informers []rbacinformers.ClusterRoleInformer
+}
+
+type mergedClusterRoleLister struct {
+	listers []rbaclisters.ClusterRoleLister
+}
+
+func (i *mergedClusterRoleInformer) Informer() cache.SharedIndexInformer {
+	panic("not implemented")
+}
+
+func (i *mergedClusterRoleInformer) Lister() rbaclisters.ClusterRoleLister {
+	aggregatedListers := make([]rbaclisters.ClusterRoleLister, 0)
+	for _, inf := range i.informers {
+		aggregatedListers = append(aggregatedListers, inf.Lister())
+	}
+	return &mergedClusterRoleLister{
+		listers: aggregatedListers,
+	}
+}
+
+func (l *mergedClusterRoleLister) List(selector labels.Selector) (ret []*rbacv1.ClusterRole, err error) {
+	aggregatedList := make([]*rbacv1.ClusterRole, 0)
+	for _, lister := range l.listers {
+		list, err := lister.List(selector)
+		if err != nil {
+			return nil, err
+		}
+		aggregatedList = append(aggregatedList, list...)
+	}
+	return aggregatedList, nil
+}
+
+func (l *mergedClusterRoleLister) Get(name string) (*rbacv1.ClusterRole, error) {
+	var errorHolder error
+	itemsToBeMerged := make([]*rbacv1.ClusterRole, 0)
+	for _, lister := range l.listers {
+		item, err := lister.Get(name)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				errorHolder = err
+			}
+		} else {
+			itemsToBeMerged = append(itemsToBeMerged, item)
+		}
+	}
+
+	if len(itemsToBeMerged) == 0 {
+		return nil, errorHolder
+	}
+
+	item := itemsToBeMerged[0]
+	for i := 1; i < len(itemsToBeMerged); i++ {
+		item.Rules = append(item.Rules, itemsToBeMerged[i].Rules...)
+	}
+
+	return item, nil
+}
+
+func (l *mergedClusterRoleLister) ListWithContext(ctx context.Context, selector labels.Selector) (ret []*rbacv1.ClusterRole, err error) {
+	aggregatedList := make([]*rbacv1.ClusterRole, 0)
+	for _, lister := range l.listers {
+		list, err := lister.ListWithContext(ctx, selector)
+		if err != nil {
+			return nil, err
+		}
+		aggregatedList = append(aggregatedList, list...)
+	}
+	return aggregatedList, nil
+}
+
+func (l *mergedClusterRoleLister) GetWithContext(ctx context.Context, name string) (*rbacv1.ClusterRole, error) {
+	var errorHolder error
+	itemsToBeMerged := make([]*rbacv1.ClusterRole, 0)
+	for _, lister := range l.listers {
+		item, err := lister.GetWithContext(ctx, name)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				errorHolder = err
+			}
+		} else {
+			itemsToBeMerged = append(itemsToBeMerged, item)
+		}
+	}
```

Comment:
why the same logic again? Just call one from the other.

Reply:
Done.